src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -778,16 +878,19 @@ const put = async (frameworkName, config, rawConfig, userToken) => {
       response = error.response;
     } else {
       // do not await for delete
+      auths.length && deleteSecret(frameworkName);
       deletePriorityClass(frameworkName);
       throw error;
     }
   }
   if (response.status !== status('Created')) {
     // do not await for delete
+    auths.length && deleteSecret(frameworkName);
```

Comment:
Do we need to delete secret in watchdog to avoid potential leak?

Reply:
better to have. only need to delete those who do not have owner.

Reply:
there exist secrets which are created elsewhere

Reply:
We can use **naming** to find job secrets created here